packages/textkit/src/types.ts

```diff
@@ -139,4 +139,17 @@ export type LayoutOptions = {
   shrinkWhitespaceFactor?: JustificationFactor;
 };
 
+export type Syllable = {
+  syllable: string;
+  hyphen?: '-';
+};
+
+export type HyphenatedWord =
+  | string[]
+  | {
+      syllables: string[] | Syllable[];
```

Comment:
Shouldn't just be this?

```suggestion
      syllables: string[];
```